ctlog/ctlog.go

```diff
@@ -26,28 +26,56 @@ import (
 	ct "github.com/google/certificate-transparency-go"
 )
 
+// Interval represents the interval [Start, End).
+type Interval struct {
+	Start, End time.Time
+}
+
 // Log contains metadata about a CT Log that is needed by Monologue.
 type Log struct {
 	Name      string
 	URL       string
 	PublicKey crypto.PublicKey
 	MMD       time.Duration
+
+	// TemporalInterval represents the interval in which a certificate's
+	// NotAfter field must fall to be accepted by the Log (as specified by the
+	// Log Operators).
+	// TemporalInterval.Start and TemporalInterval.End are both to second
+	// precision.  Any smaller units may be ignored/discarded.
+	TemporalInterval *Interval
 }
 
 // New creates a Log structure, populating the fields appropriately.
 //
 // TODO(katjoyce): replace this implementation with something less hacky that
 // takes log details from a log list struct based on the new Log list JSON
 // schema.
-func New(url, name, b64PubKey string, mmd time.Duration) (*Log, error) {
+func New(url, name, b64PubKey string, mmd time.Duration, temporalStart, temporalEnd time.Time) (*Log, error) {
 	pk, err := ct.PublicKeyFromB64(b64PubKey)
 	if err != nil {
 		return nil, fmt.Errorf("ct.PublicKeyFromB64(): %s", err)
 	}
+
+	// If only one of the temporal interval values is set, return an error.
+	if (temporalStart.IsZero() && !temporalEnd.IsZero()) ||
+		(!temporalStart.IsZero() && temporalEnd.IsZero()) {
+		return nil, fmt.Errorf("either both or neither of temporalStart and temporalEnd must be set. temporalStart = %s, temporalEnd = %s", temporalStart, temporalEnd)
+	}
```

Comment:
Could rewrite the above more concisely as:
```go
if (temporalStart.IsZero() != temporalEnd.IsZero()) {
  return nil, fmt.Errorf("either both or neither of temporalStart and temporalEnd must be set. temporalStart = %s, temporalEnd = %s", temporalStart, temporalEnd)
}
```
There's an argument that it's not as clear, but the error message does help to explain what it's checking if the reader doesn't immediately recognise that it's an XOR.

Reply:
Have removed these checks now that the parameter is an Interval instead.